docs/reference/mapping/types/rank-feature.asciidoc

```diff
@@ -58,3 +58,10 @@ the <<query-dsl-rank-feature-query,`rank_feature`>> query to modify the scoring
 in such a way that the score decreases with the value of the feature instead of
 increasing. For instance in web search, the url length is a commonly used
 feature which correlates negatively with scores.
+
+With `positive_score_impact` set to `false` for a `rank_feature` field,
+we recommend that every document that participates in a query
+has a value for this field. Otherwise, a document that has the negative feature
+would contribute to a higher score than a document that doesn't have the feature,
```

Comment:
Is there any explanation somewhere why this is so? Having read the `rank-features` docs entry, I couldn't find a reason for this. Maybe the exact details of scoring are subject to implementation details, but this behaviour with missing values is suprising. 
When I briefly played with this on example documents I was wondering what people might need to substitute in their docs in case they have a "missing" value here. Using 0 doesn't seem to work since it gives me an `illegal_argument_exception",  "reason" : "featureValue must be finite, got: Infinity for feature url_length on field _feature"` error. Maybe it is worth also explaining what to use here if values are missing.

Reply:
> Is there any explanation somewhere why this is so?

We don't have negative scores, so every query returns a score >=0 including `rank_feature` query. If used as a part of `should` clause, for documents with missing value, the `rank_feature` query will not add anything to score (0), while if a document has a feature it will return some small value .

For functions that allow features with negative score impact,  we have an explanation how scores are calculated [in this case](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-rank-feature-query.html#rank-feature-query-linear)

> I was wondering what people might need to substitute in their docs in case they have a "missing" value here.

They can't substitute a value, they just need to make sure that they use a query restrictive enough to be limited only to documents that have a value for this `rank_feature` field, for example, using a filter like : 

```js
"filter": {
  "rank_feature": {
    "field": "url_length"
  }
}      
```

Reply:
thanks for the explanation